swap/src/cli/api.rs

```diff
@@ -778,6 +819,64 @@ mod builder {
                 }
             }
 
+            // Initialize persistent EventLoop
+            if let Some(wallet) = bitcoin_wallet.clone() {
+                let namespace = XmrBtcNamespace::from_is_testnet(self.is_testnet);
+                let tor_client_for_swarm = unbootstrapped_tor_client.clone();
+                let db_for_swarm = db.clone();
+
+                let rendezvous_points = self.rendezvous_points;
+                let rendezvous_peer_ids: Vec<PeerId> =
+                    rendezvous_points.iter().map(|(p, _)| *p).collect();
+
+                let behaviour = crate::cli::Behaviour::new(
+                    env_config,
+                    wallet.clone(),
+                    seed.derive_libp2p_identity(),
+                    namespace,
+                    rendezvous_peer_ids,
+                );
+
+                let swarm = crate::network::swarm::cli(
+                    seed.derive_libp2p_identity(),
+                    tor_client_for_swarm,
+                    behaviour,
+                )
+                .await?;
+
+                let (event_loop, mut event_loop_handle) =
+                    crate::cli::EventLoop::new(swarm, db_for_swarm)?;
+
+                for (peer_id, addrs) in rendezvous_points {
+                    for addr in addrs {
+                        event_loop_handle
+                            .queue_peer_address(peer_id, addr)
+                            .await
+                            .expect(
+                                "adding a peer address will only fail if the event loop is dropped",
+                            );
+                    }
+                }
+
+                for (peer_id, addrs) in db.get_all_peer_addresses().await.context(
+                    "Failed to retrieve peer addresses from database to insert into swarm",
+                )? {
+                    for addr in addrs {
+                        event_loop_handle
+                            .queue_peer_address(peer_id, addr)
```

Comment:
we could also just use swarm.add_peer_address here instead of the channel future